assets/eip-6982/contracts/ERC721LockableUpgradeable.sol

```diff
@@ -0,0 +1,167 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+
+// Authors: Francesco Sullo <francesco@sullo.co>
+
+import "@openzeppelin/contracts-upgradeable/token/ERC721/ERC721Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC721/extensions/ERC721EnumerableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/utils/AddressUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+
+import "./IERC721Lockable.sol";
+
+contract ERC721LockableUpgradeable is
+  IERC721Lockable,
+  Initializable,
+  OwnableUpgradeable,
+  ERC721Upgradeable,
+  ERC721EnumerableUpgradeable
```

Comment:
Is the upgradeable part necessary for this reference implementation? We generally prefer minimal implementations for simplicity.